### PR TITLE
feat: new plugin allowing users to override the driver name

### PR DESCRIPTION
**Pull request by karenc-bq**
### Summary

New Connection Plugin allowing users to override the driver name.

Addresses issue #370 

### By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: wrapper/src/main/java/software/amazon/jdbc/wrapper/DatabaseMetaDataWrapper.java [hunk 1]
```diff
@@ -161,7 +161,13 @@ public String getDatabaseProductVersion() throws SQLException {
...
   @Override
   public String getDriverName() throws SQLException {
-    return DriverInfo.DRIVER_NAME;
+    return WrapperUtils.executeWithPlugins(
+        String.class,
+        SQLException.class,
+        this.pluginManager,
+        this.databaseMetaData,
+        "DatabaseMetaData.getDriverName",
+        () -> this.databaseMetaData.getDriverName());
```
Comment (sergiyvamz) [inline wrapper/src/main/java/software/amazon/jdbc/wrapper/DatabaseMetaDataWrapper.java:170]: () -> DriverInfo.DRIVER_NAME

File: wrapper/src/main/java/software/amazon/jdbc/plugin/DefaultConnectionPlugin.java [hunk 2]
```diff
@@ -95,6 +97,11 @@ public <T, E extends Exception> T execute(
 
     LOGGER.finest(
         () -> Messages.get("DefaultConnectionPlugin.executingMethod", new Object[] {methodName}));
+
+    if (GET_DRIVER_NAME.equals(methodName)) {
```
Comment (sergiyvamz) [inline wrapper/src/main/java/software/amazon/jdbc/plugin/DefaultConnectionPlugin.java:101]: I don't think we need it.

File: examples/AWSDriverExample/build.gradle.kts [hunk 3]
```diff
@@ -20,5 +20,6 @@ dependencies {
     implementation("mysql:mysql-connector-java:8.0.31")
     implementation("software.amazon.awssdk:rds:2.17.289")
     implementation("software.amazon.awssdk:secretsmanager:2.17.285")
+    implementation("com.fasterxml.jackson.core:jackson-databind:2.13.4")
```
Comment (sergiyvamz) [inline examples/AWSDriverExample/build.gradle.kts:23]: Hmm why do we need it?

**Review by sergiyvamz — APPROVED** [review]

File: examples/AWSDriverExample/build.gradle.kts (hunk 3 above)
Comment (karenc-bq) [reply examples/AWSDriverExample/build.gradle.kts:23]: Required by the secrets manager sdk

**Pull request merged**